src/qibo/tests/test_derivative.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import pytest
+
+from qibo import gates, hamiltonians
+from qibo.derivative import parameter_shift
+from qibo.models import Circuit
+
+
+# defining an observable
+def hamiltonian(nqubits=1):
+    m0 = (1 / nqubits) * hamiltonians.Z(nqubits).matrix
+    ham = hamiltonians.Hamiltonian(nqubits, m0)
+    return ham
+
+
+# defining a dummy circuit
+def circuit(nqubits=1):
+    c = Circuit(nqubits=1)
+    # all gates for which generator eigenvalue is implemented
+    c.add(gates.RX(q=0, theta=0))
+    c.add(gates.RY(q=0, theta=0))
+    c.add(gates.RZ(q=0, theta=0))
+    c.add(gates.M(0))
+    return c
+
+
+def test_derivative():
```

Comment:
In order to test all different backends, you can use the backend fixture here
```suggestion
def test_derivative(backend):
```
You don't need to define it, just put it as an argument. Its values are automatically generated from the existing conftest.py.